src/Simulation/EntryPointDriver/Azure.cs

```diff
@@ -267,10 +272,27 @@ internal sealed class AzureSettings
         /// Creates a <see cref="Workspace"/> based on the settings.
         /// </summary>
         /// <returns>The <see cref="Workspace"/> based on the settings.</returns>
-        internal Workspace CreateWorkspace() =>
-            AadToken is null
-                ? new Workspace(Subscription, ResourceGroup, Workspace, baseUri: BaseUri)
-                : new Workspace(Subscription, ResourceGroup, Workspace, AadToken, BaseUri);
+        internal Workspace CreateWorkspace()
+        {
+            if (BaseUri != null)
+            {
+                return AadToken is null
+                    ? new Workspace(Subscription, ResourceGroup, Workspace, baseUri: BaseUri)
+                    : new Workspace(Subscription, ResourceGroup, Workspace, AadToken, baseUri: BaseUri);
+            }
+            else if (Location != null)
+            {
+                return AadToken is null
+                    ? new Workspace(Subscription, ResourceGroup, Workspace, location: Location)
+                    : new Workspace(Subscription, ResourceGroup, Workspace, AadToken, location: Location);
+            }
```

Comment:
Can we make the `--base-uri` and `--location` options mutually exclusive (i.e. providing both of them is an error) rather than ignoring `--location` if `--base-uri` is provided?

Reply:
I added the validation to the `submit` command that makes `--base-uri` and `--location` mutually exclusive. Since the validation now throws an error when message when both options are included, I think it is ok to leave this logic as is since the `AzureSettings` object is kind of independent and this logic expresses the settings' precedence.

Reply:
Thanks. Can you state the precedence in the doc comments for the `BaseUri` and `Location` properties?

Reply:
Added precedence to the doc comments of both properties.